packages/connect/src/device/Device.ts

```diff
@@ -583,22 +583,8 @@ export class Device extends TypedEmitter<DeviceEvents> {
             }
         }
 
-        if (this.authenticityChecks.firmwareHash === null) {
-            this.authenticityChecks.firmwareHash = await this.checkFirmwareHash();
-        }
-
-        if (
-            // The check was not yet performed
-            this.authenticityChecks.firmwareRevision === null ||
-            // The check was performed, but outcome cannot be surely determined (Suite is offline or there was an unexpected error)
-            // -> recheck on every getFeatures() until the result is known
-            (this.authenticityChecks.firmwareRevision.success === false &&
-                ['cannot-perform-check-offline', 'other-error'].includes(
-                    this.authenticityChecks.firmwareRevision.error,
-                ))
```

Comment:
This condition is abstracted to `checkFirmwareRevisionWithRetries`. I think with named variables it doesn't need comments :slightly_smiling_face: